ibis/backends/sql/rewrites.py

```diff
@@ -385,24 +385,23 @@ def sqlize(
 
     # squash subsequent Select nodes into one
     if fuse_selects:
-        simplified = sqlized.replace(merge_select_select)
-    else:
-        simplified = sqlized
+        result = result.replace(merge_select_select)
 
     if post_rewrites:
-        simplified = simplified.replace(reduce(operator.or_, post_rewrites))
+        result = result.replace(reduce(operator.or_, post_rewrites))
 
     # extract common table expressions while wrapping them in a CTE node
-    ctes = extract_ctes(simplified)
+    ctes = extract_ctes(result)
 
-    def wrap(node, _, **kwargs):
-        new = node.__recreate__(kwargs)
-        return CTE(new) if node in ctes else new
+    if ctes:
```

Comment:
Does this save any time? Just curious. No need to dig into it either if you don't already know.

Reply:
Yes. If there are no CTEs then we avoid an entire op graph traversal, along with the logic overhead within `replace`.